@here/harp-mapview/lib/text/TextElementStateCache.ts

```diff
@@ -194,4 +233,43 @@ export class TextElementStateCache {
         this.m_referenceMap.set(textElementGroup, textElementGroupState);
         this.m_sortedGroupStates = undefined;
     }
+
+    private findDuplicate(
+        elementState: TextElementState,
+        zoomLevel: number
+    ): { entry: TextElementState[]; index: number } | undefined {
+        const element = elementState.element;
+
+        // Point labels may have duplicates (as can path labels), Identify them
+        // and keep the one we already display.
+
+        const cachedEntries = this.m_textMap.get(element.text);
+
+        if (cachedEntries === undefined) {
+            // No labels found with the same text.
+            return undefined;
+        }
+
+        const maxSqDistError = getDedupSqDistTolerance(zoomLevel);
+
+        // Other labels found with the same text. Check if they're near enough to be considered
+        // duplicates.
+        const entryCount = cachedEntries.length;
+        const elementPosition = element.points as THREE.Vector3;
+        let duplicateIndex: number = -1;
+        for (let i = 0; i < entryCount; ++i) {
```

Comment:
how do you prevent this checking against itself? as far as i can read, the TextElementState is part of this cache, so it is possible that the element is checked to see if it is a duplicate with itself?

Reply:
The duplicate cache is cleared after every frame and refilled again in a loop that iterates once through  every visited label. See the call to `deduplicateElement` in `prepareTextElementGroup`. So every visited label is added at most once per frame.